knative-operator/pkg/controller/knativeserving/consoleclidownload/consoleclidownload.go

```diff
@@ -158,87 +154,62 @@ func deleteDeprecatedResources(instance *servingv1alpha1.KnativeServing, apiclie
 	return nil
 }
 
-// makeKnService makes Knative Service object and its SPEC from provided image parameter
-func makeKnService(image string, instance *servingv1alpha1.KnativeServing) *servingv1.Service {
-	if instance == nil {
-		return nil
-	}
-	// OwnerReference is not used to handle ksvc cleanup due to race condition with control-plane deletion.
-	// In such a case route's finalizer blocks clean removal of resources.
-	service := &servingv1.Service{
+func makeRoute(instance *servingv1alpha1.KnativeServing) *routev1.Route {
+	return &routev1.Route{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      knConsoleCLIDownloadService,
-			Namespace: instance.Namespace,
+			Name:      knCLIDownload,
+			Namespace: os.Getenv("NAMESPACE"),
 			Annotations: map[string]string{
-				common.ServingOwnerName:      instance.Name,
-				common.ServingOwnerNamespace: instance.Namespace,
+				common.ServingOwnerName:      instance.GetName(),
+				common.ServingOwnerNamespace: instance.GetNamespace(),
 			},
 		},
-		Spec: servingv1.ServiceSpec{
-			ConfigurationSpec: servingv1.ConfigurationSpec{
-				Template: servingv1.RevisionTemplateSpec{
-					Spec: servingv1.RevisionSpec{
-						PodSpec: corev1.PodSpec{
-							Containers: []corev1.Container{
-								{
-									Image: image,
-									Resources: corev1.ResourceRequirements{
-										Requests: corev1.ResourceList{
-											corev1.ResourceCPU:    resource.MustParse("10m"),
-											corev1.ResourceMemory: resource.MustParse("50Mi"),
-										},
-									},
-								},
-							},
-						},
-					},
-				},
+		Spec: routev1.RouteSpec{
+			To: routev1.RouteTargetReference{
+				Kind: "Service",
+				Name: "knative-openshift-metrics-3",
```

Comment:
non straightforward name :thinking: we might want to consider renaming the service or adding a new one.

Reply:
@markusthoemmes I would just add a new one to separate things because these resources are usually tight to the corresponding deployment (onwener refs). Btw we had that name because it is used by the `knative-openshift` deployment plus the suffix for the migration issues we faced with OLM.  Why is this route to the service monitor service being created?

Reply:
FWIW, the lifecycle of this is managed by OLM, not by the deployment. This is not one of the autoimatically injected services.

The route is created to make the CLI artifact server available for download via the ConsoleCLIDownload resource.

Reply:
Yes it is not I am saying when I would add a new one. Ok from what I see you take advantage of that service to get access to the container for the cli. It seems to should work, cant thing of a problem right now. The service monitor access only the metrics port so no collision there.